### PR TITLE
Add Lazy Execution Option in Legacy Compile Path

### PR DESCRIPTION
**Pull request by pglusacTT**
### Ticket
N/A

### Problem description
Legacy path always forces torch_xla sync. In training, we want more freedom to target specific tensors (like gradients and loss only) in order to produce cleaner execution graphs.

### What's changed
- Add `tt_lazy_execution` flag that is off by default and it skips the syncs in the `__call__` method.

### Checklist
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by pglusacTT** [comment]
FYI @abogdanovicTT @agobeljicTT

File: python_package/tt_torch/backend/backend.py [hunk 1]
```diff
@@ -242,6 +244,9 @@ def __call__(self, *args):
                 gm_has_functional_output_kind = False
                 break
 
+        if self.lazy_execution:
+            return output
+
```
Comment (agobeljicTT) [inline python_package/tt_torch/backend/backend.py]: what if `gm_has_functional_output_kind`
does the user need to manually do `torch_xla._XLAC._xla_sync_multi(output_tensors, devices, wait=False)` afterwards or?

Comment (pglusacTT) [reply python_package/tt_torch/backend/backend.py]: I've moved the early exit upwards so it doesn't compute `gm_has_functional_output_kind`.
Sync is left to the user, which we already do in our training scripts.

**Review by mstojkovicTT — APPROVED** [review]
lgtm!

**Review by vkovinicTT — APPROVED** [review]
